### PR TITLE
feat: support `default_thread_rate_limit_per_user` in channel creation

### PR DESCRIPTION
**Pull request by jaw0r3k**
**Please describe the changes this PR makes and why it should be merged:**
Adds:
- #9273

##### Summary
🧵🐌🛠️

This pull request adds support for setting a default slowmode for threads in guild channels. It introduces a new property `defaultThreadRateLimitPerUser` to the `GuildChannelManager`, `BaseGuildTextChannel`, and `CategoryChannel` classes, and updates the channel creation methods and documentation accordingly.

##### Walkthrough
*  Add `defaultThreadRateLimitPerUser` property to guild channel classes and options ([link](https://github.com/discordjs/discord.js/pull/9339/files?diff=unified&w=0#diff-ac36f82cbe3c9571b4d973edf8122af9059492c405b321ad8bcb60aacc988d37R153), [link](https://github.com/discordjs/discord.js/pull/9339/files?diff=unified&w=0#diff-ac36f82cbe3c9571b4d973edf8122af9059492c405b321ad8bcb60aacc988d37R195), [link](https://github.com/discordjs/discord.js/pull/9339/files?diff=unified&w=0#diff-42032b879291eabbae9a6611057c6e24a618834d0f048402192606214a874382R77-R86), [link](https://github.com/discordjs/discord.js/pull/9339/files?diff=unified&w=0#diff-fb00b4bd9fa501897e5ad1e7858c5c0c9d5c1899acbe1ebb80bcb0f21a64916cR52-R53))

**Status and versioning classification:**
- Code changes have been tested against the Discord API
- I know how to update typings and have done so
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

**Review by SpaceEEC — CHANGES_REQUESTED** [review]
Looking good.

---

What's with the raw data types however? I feel like they don't belong into this PR :eyes:

**Comment by jaw0r3k** [comment]
> Looking good.
> 
> What's with the raw data types however? I feel like they don't belong into this PR 👀

There were no other way to add it to this pr

**Comment by SpaceEEC** [comment]
I don't understand why these changes were made, your changes in `index.d.ts` do not refer to any of those.

**Comment by jaw0r3k** [comment]
> I don't understand why these changes were made, your changes in `index.d.ts` do not refer to any of those.

Oh yeah other channel objects didn't have to be updated, but they weren't so w/e

File: typings/rawDataTypes.d.ts [hunk 1]
```diff
@@ -279,3 +290,52 @@ export interface APIApplicationRoleConnectionMetadata {
   description: string;
   description_localizations?: LocalizationMap;
 }
+
+export interface APITextBasedChannel<T extends ChannelType> extends APIChannelBase<T> {
```
Comment (vladfrangu) [inline typings/rawDataTypes.d.ts:294]: Can you please make this change in a different PR? Its not really tied to this PR

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: typings/rawDataTypes.d.ts (hunk 1 above)
Comment (jaw0r3k) [reply typings/rawDataTypes.d.ts:294]: if you really want to, okey

**Comment by Jiralite** [comment]
This needs a rebase.

**Review by Jiralite — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]
